lib/services/android-project-service.ts

```diff
@@ -287,10 +288,14 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 				if (this.$hostInfo.isWindows) {
 					gradleBin += ".bat"; // cmd command line parsing rules are weird. Avoid issues with quotes. See https://github.com/apache/cordova-android/blob/master/bin/templates/cordova/lib/builders/GradleBuilder.js for another approach
 				}
-				this.spawn(gradleBin, buildOptions, { stdio: "inherit", cwd: this.platformData.projectRoot }).wait();
+
+				try {
+					this.spawn(gradleBin, buildOptions, { stdio: "inherit", cwd: this.platformData.projectRoot }).wait();
+				} catch (ex) {
+					this.$errors.failWithoutHelp("Gradle build failed." + EOL + platformRemoveMessage);
```

Comment:
this will hide the real error from us and from the user. At least add
```TypeScript
this.$logger.trace("Error while executing gradle build: ", ex);
```
before the `this.$errors.fail...`
As debugging this will be painful without real error.